api/lambdas/cda-to-visualization/index.js

```diff
@@ -0,0 +1,217 @@
+    const url = await convertStoreAndReturnHtmlDocUrl({ fileName, document });
+    console.log("html", url);
+    return url;
+  }
+
+  if (conversionType === "pdf") {
+    const url = await convertStoreAndReturnPdfDocUrl({ fileName, document });
+    console.log("pdf", url);
+    return url;
+  }
+
+  return;
+};
+
+const downloadDocumentFromS3 = async ({ fileName }) => {
+  const file = await s3client
+    .getObject({
+      Bucket: bucketName,
+      Key: fileName,
+    })
+    .promise();
+
+  const data = file.Body?.toString("utf-8");
+
+  return data;
+};
+
+const convertStoreAndReturnHtmlDocUrl = async ({ fileName, document }) => {
+  const convertDoc = await convertToHtml(document);
+
+  const newFileName = fileName.concat(".html");
+
+  await s3client
+    .putObject({
+      Bucket: bucketName,
+      Key: newFileName,
+      Body: convertDoc.toString(),
+      ContentType: "text/html",
+    })
+    .promise();
+
+  const urlHtml = await getSignedUrl({ fileName: newFileName });
+
+  return urlHtml;
+};
+
+const convertStoreAndReturnPdfDocUrl = async ({ fileName, document }) => {
+  const convertDoc = await convertToHtml(document);
+
+  const htmlFilepath = `/tmp/${fileName}`;
+
+  fs.writeFileSync(htmlFilepath, convertDoc);
+
+  // Defines filename + path for downloaded HTML file
+  const pdfFilename = fileName.concat(".pdf");
+  const pdfFilepath = `/tmp/${pdfFilename}`;
+
+  // Defines URL to read htmlFilepath
+  const fetchUrl = `file://${htmlFilepath}`;
+
+  // Define
+  let browser = null;
+
+  try {
+    // Defines browser
+    browser = await puppeteer.launch({
+      args: chromium.args,
+      defaultViewport: chromium.defaultViewport,
+      executablePath: await chromium.executablePath,
+      headless: chromium.headless,
+    });
+
+    // Defines page
+    let page = await browser.newPage();
+
+    // Navigate to page, wait until dom content is loaded
+    await page.goto(fetchUrl, {
+      waitUntil: "domcontentloaded",
+    });
+
+    // Wait 2.5 seconds
+    await delay(2500);
+
+    // Generate PDF from page in puppeteer
+    await page.pdf({
+      path: pdfFilepath,
+      printBackground: true,
+      format: "A4",
+      margin: {
+        top: "20px",
+        left: "20px",
+        right: "20px",
+        bottom: "20px",
+      },
+    });
+
+    // Upload generated PDF to S3 bucket
+    await new Promise((resolve, reject) => {
+      s3client
+        .upload({
+          Bucket: bucketName,
+          Key: pdfFilename,
+          Body: fs.readFileSync(pdfFilepath),
+        })
+        .send((err, data) => {
+          console.log(err, data);
+          // Logs error
+          if (err) {
+            console.log(`generate-pdf -> upload to s3 -> ERROR`);
+            console.log(err);
+            reject(err);
+            return;
+          }
+          console.log(`generate-pdf -> upload to s3 -> SUCCESS --> ${htmlFilepath}`);
+          resolve(true);
+        });
+    });
```

Comment:
Why a diff approach to upload to S3? It doesn't seem like we're sending a stream or anything, ideally we would use the same approach as the one on `convertStoreAndReturnHtmlDocUrl()`, could even have a shared function for it.